contracts/src/L1/rollup/ScrollChain.sol

```diff
@@ -353,6 +353,63 @@ contract ScrollChain is OwnableUpgradeable, PausableUpgradeable, IScrollChain {
         emit FinalizeBatch(_batchIndex, _batchHash, _postStateRoot, _withdrawRoot);
     }
 
+    /// @inheritdoc IScrollChain
+    function finalizeBatch(
```

Comment:
Could we add a `block.chainid != 1` check to double protect that the method cannot be called in Mainnet?

Reply:
good point. Let's add this check

Reply:
This implementation never deployed to mainnet, it is not needed.

Reply:
Yes. Thus I mentioned that it's just a "double protection".

Reply:
Agree with @colinlyguo. The additional check just prevents this being accidentally deployed to mainnet though it shouldn't happen